app/src/main/java/br/org/cesar/knot_setup_app/activity/configureDevice/ConfigureDevicePresenter.java

```diff
@@ -39,9 +46,82 @@
         this.bluetoothWrapper = KnotSetupApplication.getBluetoothWrapper();
         this.device = KnotSetupApplication.getBluetoothDevice();
         this.context = context;
+        Log.d("DEV-LOG","getOpenthreadConfgig starting");
+        getOpenthreadConfig();
+    }
+
+
+    private void getOpenthreadConfig(){
+        String bearer,ip,port,request;
+        bearer = dataManager.getInstance()
+                .getPersistentPreference()
+                .getSharedPreferenceString(context,"token");
+
+        ip = dataManager.getInstance()
```

Comment:
Why are you getting ip and port again if you already have it from the gateway?

Reply:
Instead of passing them as arguments from view to view, I chose to store them as user preferences and get them from any view that needs it.